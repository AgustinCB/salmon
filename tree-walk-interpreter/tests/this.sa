class Cake {
  taste() {
    var adjective = "delicious";
    print "The " + this.flavor + " cake is " + adjective + "!";
  }

  getCallback() {
    fun localFunction() {
      print this.flavor;
    }

    return localFunction;
  }
}

var cake = Cake();
cake.flavor = "German chocolate";
cake.taste();
var cb = cake.getCallback();
cb();
print this;