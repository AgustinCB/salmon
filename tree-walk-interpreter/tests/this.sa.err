There was an error! [file stdin line 21] Error: Variable `this` not found!
