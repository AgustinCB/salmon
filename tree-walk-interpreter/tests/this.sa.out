The German chocolate cake is delicious!
German chocolate
